napari/layers/points/points.py

```diff
@@ -275,6 +279,17 @@ def __init__(
             properties = dataframe_to_properties(properties)
         self._properties = self._validate_properties(properties)
 
+        if default_properties is None:
+            default_properties = {}
+        elif not isinstance(default_properties, dict):
+            default_properties = dataframe_to_properties(default_properties)
+        self._default_properties = default_properties
+
+        if len(data) == 0 and self.properties and not default_properties:
+            raise ValueError(
```

Comment:
Do we also want to check the actual keys from `properties` are present in `default_properties`? Should you be forced to provide a default for every property in this case? I can trigger errors with this combo say
```python
    # create properties for each point
    properties = {
        'class': np.array([]),
    }
    default_properties = {'label': [0, 1, 2]}
```
might be better to catch them here first?

Reply:
Good point! I added a check and a test for the check.